### PR TITLE
interpreters/duktape:Add duktape to global search path

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary

This is useful for using duktape as a library.

## Impact
None

## Testing
Tested on sim & stm32

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
let's use the method simliar with lvgl instead?

**Comment by no1wudi** [comment]
> let's use the method simliar with lvgl instead?
But the app want to use Duktape must add include path to CFLAGS manually.

File: interpreters/duktape/Makefile [hunk 1]
```diff
@@ -48,23 +48,31 @@ CFLAGS += -I$(DUKTAPE_UNPACK)/extras/print-alert
...
-	@echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
-	@tar xvfJ $(DUKTAPE_TARBALL)
-	@echo "Patching $(DUKTAPE_UNPACK)"
-	@patch -p0 < duk_cmdline.patch
-	@touch $(DUKTAPE_UNPACK)/.patch
+	$(Q) echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
+	$(Q) tar xvfJ $(DUKTAPE_TARBALL)
+	$(Q) ln -sf $(DUKTAPE_UNPACK) duktape
+	$(Q) ln -sf $(APPINC)/src-noline/duktape.h $(APPINC)/duktape.h
+	$(Q) ln -sf $(APPINC)/src-noline/duk_config.h $(APPINC)/duk_config.h
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:62]: don't need link header file

File: interpreters/duktape/Makefile [hunk 2]
```diff
@@ -48,23 +48,31 @@ CFLAGS += -I$(DUKTAPE_UNPACK)/extras/print-alert
...
 
 $(DUKTAPE_UNPACK): $(DUKTAPE_TARBALL)
-	@echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
-	@tar xvfJ $(DUKTAPE_TARBALL)
-	@echo "Patching $(DUKTAPE_UNPACK)"
-	@patch -p0 < duk_cmdline.patch
-	@touch $(DUKTAPE_UNPACK)/.patch
+	$(Q) echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
+	$(Q) tar xvfJ $(DUKTAPE_TARBALL)
+	$(Q) ln -sf $(DUKTAPE_UNPACK) duktape
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:60]: let's define DUKTAPE_UNPACK=duktape and mv duktape-2.5.0 to duktape directly instead link

File: interpreters/duktape/Makefile [hunk 3]
```diff
@@ -48,23 +48,30 @@ CFLAGS += -I$(DUKTAPE_UNPACK)/extras/print-alert
...
 $(DUKTAPE_UNPACK): $(DUKTAPE_TARBALL)
-	@echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
-	@tar xvfJ $(DUKTAPE_TARBALL)
-	@echo "Patching $(DUKTAPE_UNPACK)"
-	@patch -p0 < duk_cmdline.patch
-	@touch $(DUKTAPE_UNPACK)/.patch
+	$(Q) echo "Unpacking $(DUKTAPE_TARBALL) to $(DUKTAPE_UNPACK)"
+	$(Q) tar xvfJ $(DUKTAPE_TARBALL)
+	$(Q) mv duktape-$(DUKTAPE_VERSION) $(DUKTAPE_UNPACK)
+	$(Q) ln -sf $(APPINC)/src-noline/duktape.h $(APPINC)/duktape.h
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:61]: change all $(APPINC) to $(DUKTAPE_UNPACK)?

Comment (no1wudi) [reply interpreters/duktape/Makefile]: $(DUKTAPE_UNPACK) is a relative path, may cause some problem.

File: interpreters/duktape/Makefile [hunk 4]
```diff
@@ -48,23 +48,28 @@ CFLAGS += -I$(DUKTAPE_UNPACK)/extras/print-alert
...
 $(DUKTAPE_UNPACK)/.patch: $(DUKTAPE_UNPACK)
+	$(Q) touch $(DUKTAPE_UNPACK)/.patch
 
 context:: $(DUKTAPE_UNPACK)/.patch
 
 distclean::
 	$(call DELDIR, $(DUKTAPE_UNPACK))
 	$(call DELFILE, $(DUKTAPE_TARBALL))
+	$(call DELFILE, duktape.h)
+	$(call DELFILE, duk_config.h)
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:73]: remove, don't need.

File: interpreters/duktape/Makefile [hunk 5]
```diff
@@ -48,23 +48,28 @@ CFLAGS += -I$(DUKTAPE_UNPACK)/extras/print-alert
 CFLAGS += -DDUK_CMDLINE_CONSOLE_SUPPORT
 CFLAGS += -DDUK_CMDLINE_PRINTALERT_SUPPORT
 
+DUKINC = $(CURDIR)/$(DUKTAPE_UNPACK)
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Makefile:51]: remove, no user.

File: interpreters/duktape/.gitignore [hunk 6]
```diff
@@ -1 +1 @@
-duktape-*
+duktape*
```
Comment (xiaoxiang781216) [inline interpreters/duktape/.gitignore:1]: add newline

**Comment by xiaoxiang781216** [comment]
LGTM.

**Pull request merged**